Code/Tools/AssetProcessor/native/AssetManager/assetProcessorManager.cpp

```diff
@@ -3782,33 +3802,114 @@ namespace AssetProcessor
+                    newDependencies.push_back(AZStd::move(newDependencyEntry));
+                }
+            }
+        }
+
+        AZStd::unordered_set<AZStd::string> resolvedSourceDependenciesDeduplication;
+
+        for (const AZStd::pair<AZ::Uuid, AssetBuilderSDK::SourceFileDependency>& sourceDependency : entry.m_sourceFileDependencies)
+        {
+            // figure out whether we can resolve the dependency or not:
+            QStringList resolvedDependencyList;
+            QString resolvedDatabaseName;
+            if (!ResolveSourceFileDependencyPath(sourceDependency.second, resolvedDatabaseName, resolvedDependencyList))
+            {
+                // ResolveDependencyPath should only fail in a data error, otherwise it always outputs something,
+                // even if that something starts with the placeholder.
+                continue;
+            }
+
+            // Handle multiple resolves (wildcard dependencies)
+            for (const auto& thisEntry : resolvedDependencyList)
+            {
+                if(jobDependenciesDeduplication.contains(DependencyDeduplication{sourceDependency.first, entry.m_sourceFileInfo.m_databasePath.toUtf8().constData(),
+                    thisEntry.toUtf8().constData()}))
+                {
+                    for (JobDetails& job : entry.m_jobsToAnalyze)
+                    {
+                        job.m_warnings.push_back(AZStd::string::format(
+                            "Builder `%s` emitted Source Dependency and Job Dependency on file `%s`.  "
+                            "This is unecessary and the builder should be updated to only emit the Job Dependency.",
+                            job.m_assetBuilderDesc.m_name.c_str(), thisEntry.toUtf8().constData()));
+                    }
+
+                    continue;
+                }
+
+                // Sometimes multiple source dependencies can resolve to the same file due to the overrides system
+                // Eliminate the duplicates, no warning is needed since the builder can't be expected to handle this
+                if(auto result = resolvedSourceDependenciesDeduplication.insert(thisEntry.toUtf8().constData()); result.second)
+                {
+                    // add the new dependency:
+                    SourceFileDependencyEntry newDependencyEntry(
+                        sourceDependency.first,
                         entry.m_sourceFileInfo.m_databasePath.toUtf8().constData(),
                         thisEntry.toUtf8().constData(),
-                        SourceFileDependencyEntry::DEP_JobToJob,   // significant line in this code block
+                        SourceFileDependencyEntry::DEP_SourceToSource,
                         false);
                     newDependencies.push_back(AZStd::move(newDependencyEntry));
                 }
+            }
 
+            if (jobDependenciesDeduplication.contains(DependencyDeduplication{
```

Comment:
you may want to format this more vertically to make it more readable and match the other formats in this file